akka-cluster/src/test/scala/akka/cluster/ClusterSpec.scala

```diff
@@ -145,6 +151,22 @@ class ClusterSpec extends AkkaSpec(ClusterSpec.config) with ImplicitSender {
       cluster.remotePathOf(testActor).address should ===(selfAddress)
     }
 
+    "log a message when becoming a leader and when stop being a leader" in {
+
+      EventFilter
+        .info(occurrences = 2, pattern = "((^|, )(.+?(is the new leader)|.+?(is no longer the leader)))+$")
```

Comment:
I think this can be done with two separate checks. The first around `leaderActions() // Joining -> Up` and the second around the last `leaderActions()`

Reply:
This has been changed in the latest commit.